src/shogun/base/SGObject.h

```diff
@@ -100,6 +103,16 @@ template <class T> class SGStringList;
 
 #define SG_ADD(...) VARARG(SG_ADD, __VA_ARGS__)
 
+#define VALUE_TO_STRING_MACRO(s) #s
+
+#define SG_ADD_OPTION(param_name, enum_value)                                  \
+	{                                                                          \
+		static_assert(                                                         \
+		    std::is_enum<decltype(enum_value)>::value, "Expected an enum!");   \
+		m_string_to_enum_map[param_name][VALUE_TO_STRING_MACRO(enum_value)] =  \
+		    enum_value;                                                        \
+	}
+
```

Comment:
pls undefine your helper macro after having used it

Reply:
Unfortunately I can't :( the macro is needed by `SG_ADD_OPTION`, which is used later on

Reply:
I see, well ok